jac/jaclang/compiler/absyntree.py

```diff
@@ -540,9 +540,9 @@ def sem_token(self) -> Optional[tuple[SemTokType, SemTokMod]]:
 class ArchSpec(ElementStmt, CodeBlockStmt, AstSymbolNode, AstDocNode, AstSemStrNode):
     """ArchSpec node type for Jac Ast."""
 
-    def __init__(self, decorators: Optional[SubNodeList[Expr]] = None) -> None:
+    def __init__(self, decorators: Optional[list[Expr]] = None) -> None:
```

Comment:
Don't make it an optional list, just `list[Expr]` if there is no decorators the list will be empty

Reply:
That case all the places we're doing
```python
if self.decorators:
  for decor in self.decorators:
    process(decor)
```

will become:
```python
for decor in self.decorators:
  process(decor)
```